apps/api/tests/api/application-state.test.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { PostgreSqlContainer, StartedPostgreSqlContainer } from '@testcontainers/postgresql';
+import assert from 'node:assert';
+import { after, before, describe, it } from 'node:test';
+
+import { ApplicationStates, ApplicationStateValues } from '@pcgl-daco/data-model/src/types.js';
+import { ApplicationStateManager, createApplicationStateManager } from '../../src/api/states.js';
+import { connectToDb, type PostgresDb } from '../../src/db/index.js';
+import { addInitialApplications, initTestMigration, PG_DATABASE, PG_PASSWORD, PG_USER } from '../testUtils.js';
+
+const {
+	APPROVED,
+	CLOSED,
+	DAC_REVIEW,
+	DAC_REVISIONS_REQUESTED,
+	DRAFT,
+	INSTITUTIONAL_REP_REVIEW,
+	REJECTED,
+	REP_REVISION,
+	REVOKED,
+} = ApplicationStates;
+
+describe('State Machine', () => {
+	let db: PostgresDb;
+	let container: StartedPostgreSqlContainer;
+	let testStateManager: ApplicationStateManager;
+
+	before(async () => {
+		container = await new PostgreSqlContainer()
+			.withUsername(PG_USER)
+			.withPassword(PG_PASSWORD)
+			.withDatabase(PG_DATABASE)
+			.start();
+
+		const connectionString = container.getConnectionUri();
+		db = connectToDb(connectionString);
+
+		await initTestMigration(db);
+		await addInitialApplications(db);
+	});
+
+	describe('Application State Manager', () => {
+		let stateValue: ApplicationStateValues = DRAFT;
+
+		it('create Application State Manager classes', async () => {
+			const result = await createApplicationStateManager({ id: 1 });
+			assert.ok(result.success);
+
+			testStateManager = result.data;
+			assert.ok(testStateManager);
+		});
+
+		it('should initialize with state DRAFT', () => {
+			stateValue = testStateManager.getState();
+			assert.strictEqual(stateValue, DRAFT);
+		});
```

Comment:
'Happy path' tests are reusing the same `testStateManager` and updating it from one test to the next. If preferred I can update this to create a new Class on every test, so each test is standalone.